app/screens/ChildSetup.tsx

```diff
@@ -183,16 +187,30 @@ const ChildSetup = ({ navigation }: Props): any => {
   };
   const importFromFile = async (): Promise<any> => {
     if (Platform.OS == "android") {
-      const dataset = await ScopedStorage.openDocument(true, 'base64');
-      console.log(dataset, "..dataset");
+      const dataset = await ScopedStorage.openDocument(true, 'utf8');
+      let oldChildrenData: any = null;
+      let importedrealm: any = null;
       if (dataset && dataset.data != "" && dataset.data != null && dataset.data != undefined) {
-        await RNFS.writeFile(tempRealmFile, dataset.data, "base64");
-        const importedrealm = await new Realm({ path: 'user1.realm' });
-        //console.log(importedrealm, "...importedrealm");
-        const user1Path = importedrealm.path;
-        console.log(user1Path, "..user1Path");
-        const oldChildrenData = importedrealm.objects('ChildEntity');
-        //console.log(oldChildrenData, "..newoldChildrenData..")
+        if (dataset.name.endsWith('.json')) {
+          const decryptedData = AesCrypto.decrypt(dataset.data, encryptionsKey, encryptionsIVKey, 'aes-256-cbc')
+            .then((text: any) => {
+              return text;
+            })
+            .catch((error: any) => {
+              console.log("Decrypted error", error);
+              throw error;
+            });
+          const importedJsonData = JSON.parse(await decryptedData);
+          await RNFS.writeFile(tempRealmFile, JSON.stringify(decryptedData), "utf8");
+          oldChildrenData = importedJsonData;
+        } else {
```

Comment:
Should we validate file types? How does app behave if user selects inappropriate file?